src/common/Requirements.ts

```diff
@@ -123,82 +119,87 @@ export abstract class BaseSetup implements RequirementList {
     }
 
     public async executeSetup(): Promise<SetupTestResult> {
-        const allPromises: Array<Promise<any>> = [];
-        this.requirements.forEach((requirement) =>
-            allPromises.push(WrappedPromise(requirement))
+        const testResult: SetupTestResult = {
+            hasMetAllRequirements: true,
+            tests: []
+        };
+
+        let totalDuration = 0;
+        const setupTasks = new Listr(
```

Comment:
create a `Listr` object for `Setup` which itself holds another `Lister` object for all of the subtasks (resulting in auto-indentation when printing output)